src/lib.rs

```diff
@@ -191,16 +189,14 @@ where
 
     pub fn read_raw(&mut self) -> Result<(i16, i16, i16), Error<E>> {
```

Comment:
We just split the check overflow process, which is required for every reading.

Now Itsn't necessary write two times on i2c, the reading sensors and the ST2, 
as ST2 is the next after the reading bytes.